server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -123,6 +133,34 @@ public ClusterStateHealth(final ClusterState clusterState, final String[] concre
             computeStatus = getClusterHealthStatus(indexHealth, computeStatus);
         }
 
+        if (hasSearchOnlyIndices) {
+            Map<String, ClusterIndexHealth> searchOnlyIndices = new HashMap<>();
+            for (String index : concreteIndices) {
+                IndexMetadata indexMetadata = clusterState.metadata().index(index);
+                if (indexMetadata == null) continue;
+
+                boolean isSearchOnlyEnabled = indexMetadata.getSettings()
```

Comment:
```suggestion
                boolean isSearchOnlyCluserBlockEnabled = indexMetadata.getSettings()
```

I think here and elsewhere we need to be clear to differentiate between "The index is search only", ie has search only shards and the index has the cluster block applied.

Reply:
Sure let me find and update to `isSearchOnlyCluserBlockEnabled` (during the setting check `INDEX_BLOCKS_SEARCH_ONLY_SETTING`.)

Reply:
I have updated this in my latest commit, please check.